irmaclient/sdjwtvc_storage.go

```diff
@@ -365,28 +368,42 @@ func getFirstCredentialInstanceFromBucket(bucket *bbolt.Bucket, aesKey [32]byte)
 	return sdjwtvc.SdJwtVc(decrypted), nil
 }
 
-func getCredentialInfoFromBucket(bucket *bbolt.Bucket, aesKey [32]byte) (*irma.CredentialInfo, error) {
+func getCredentialInfoFromBucket(bucket *bbolt.Bucket, aesKey [32]byte) (*SdJwtVcBatchMetadata, error) {
 	encrypted := bucket.Get([]byte(infoKey))
 	decrypted, err := decrypt(encrypted, aesKey)
 	if err != nil {
 		return nil, fmt.Errorf("failed to decrypt: %v", err)
 	}
 
-	var info irma.CredentialInfo
+	var info SdJwtVcBatchMetadata
 	err = json.Unmarshal(decrypted, &info)
 	if err != nil {
 		return nil, fmt.Errorf("failed to unmarshal: %v (%v)", err, string(decrypted))
 	}
 
-	return &info, nil
+	instanceCount, err := getCredentialInstanceCount(bucket)
+
+	if err != nil {
+		return nil, fmt.Errorf("failed to get instance count: %v", err)
+	}
+
+	return &SdJwtVcBatchMetadata{
```

Comment:
Why create a new struct here instead of only setting the `RemainingInstanceCount` property on the `info` object and returning a pointer to that instance?